esprima.js

```diff
@@ -1402,6 +1542,11 @@
             return scanStringLiteral();
         }
 
+        // Template literals start with ` (U+0060) for template head
+        // or } (U+007D) for template middle or template tail.
+        if (ch === 0x60 || (ch === 0x7D && state.curlyStack[state.curlyStack.length - 1] === '${')) {
+            return scanTemplate();
+        }
```

Comment:
I think template literals are not so common yet. Thus, better to have this check as the last one (before `return scanPunctuator()`) for minor performance gain.